### PR TITLE
v3: added ref type to the `Link` component

### LINKED ISSUE #341: Links don't support `ref` type
The `ref` type is missing on a `Link` component.

### PR DESCRIPTION
**Pull request by jeetiss**
closes #341

### CODE REVIEW

**Comment by molefrog** [comment]
Keep in mind, `Link` can wrap custom components as well:

```jsx
<Link>
  <MyCustomLinkComponent />
</Link>
```

A ref to `MyCustomLinkComponent` isn't always `HTMLAnchorElement`.

File: packages/wouter/types/index.d.ts [hunk 1]
```diff
@@ -113,7 +114,7 @@ export function Redirect<H extends BaseLocationHook = LocationHook>(
 ): ReactElement<any, any> | null;
 
 export function Link<H extends BaseLocationHook = LocationHook>(
-  props: PropsWithChildren<LinkProps<H>>,
+  props: PropsWithChildren<LinkProps<H>> & RefAttributes<HTMLAnchorElement>,
```
Comment (molefrog) [inline packages/wouter/types/index.d.ts:117]: My guess is, you can try to infer the type of `children` prop and then use `ElementRef<props["children"]>`

Comment (jeetiss) [reply packages/wouter/types/index.d.ts:117]: it doesn't work

and it makes no sense to pass ref object to children:

```jsx
<Link ref={ref}>
  <MyCustomLinkComponent />
</Link>
```

should be 👇🏻👇🏻👇🏻
```jsx
<Link>
  <MyCustomLinkComponent ref={ref} />
</Link>
```

in userland

**Review by molefrog — APPROVED** [review]
LGTM

**Pull request merged**